In [5]:
from tabl import *

# THE BASICS

CapitalizedWords for term T(n, k) (type int).

In [6]:
StirlingSet(5, 3)

25

All lower case for row T(n) (type list[int]). 

In [7]:
stirlingset(5)

[0, 1, 15, 25, 10, 1]

In [19]:
StirlingSet.tab(9)

[[1],
 [0, 1],
 [0, 1, 1],
 [0, 1, 3, 1],
 [0, 1, 7, 6, 1],
 [0, 1, 15, 25, 10, 1],
 [0, 1, 31, 90, 65, 15, 1],
 [0, 1, 63, 301, 350, 140, 21, 1],
 [0, 1, 127, 966, 1701, 1050, 266, 28, 1]]

In [12]:
StirlingSet.rev(9)

[[1],
 [1, 0],
 [1, 1, 0],
 [1, 3, 1, 0],
 [1, 6, 7, 1, 0],
 [1, 10, 25, 15, 1, 0],
 [1, 15, 65, 90, 31, 1, 0],
 [1, 21, 140, 350, 301, 63, 1, 0],
 [1, 28, 266, 1050, 1701, 966, 127, 1, 0]]

In [13]:
StirlingSet.inv(9)

[[1],
 [0, 1],
 [0, -1, 1],
 [0, 2, -3, 1],
 [0, -6, 11, -6, 1],
 [0, 24, -50, 35, -10, 1],
 [0, -120, 274, -225, 85, -15, 1],
 [0, 720, -1764, 1624, -735, 175, -21, 1],
 [0, -5040, 13068, -13132, 6769, -1960, 322, -28, 1]]

'stirling_set.mat' converts the triangle to a square matrix by filling T(n, k) for k > n with zeros.

In [14]:
StirlingSet.mat(9)

[[1, 0, 0, 0, 0, 0, 0, 0, 0],
 [0, 1, 0, 0, 0, 0, 0, 0, 0],
 [0, 1, 1, 0, 0, 0, 0, 0, 0],
 [0, 1, 3, 1, 0, 0, 0, 0, 0],
 [0, 1, 7, 6, 1, 0, 0, 0, 0],
 [0, 1, 15, 25, 10, 1, 0, 0, 0],
 [0, 1, 31, 90, 65, 15, 1, 0, 0],
 [0, 1, 63, 301, 350, 140, 21, 1, 0],
 [0, 1, 127, 966, 1701, 1050, 266, 28, 1]]

The identifier of the function. 

In [15]:
StirlingSet.id

'StirlingSet'

# CORE TABLES

In [9]:
t = StirlingSet.tab(9)
PrintFlat(t)

[[1], [0, 1], [0, 1, 1], [0, 1, 3, 1], [0, 1, 7, 6, 1], [0, 1, 15, 25, 10, 1], [0, 1, 31, 90, 65, 15, 1], [0, 1, 63, 301, 350, 140, 21, 1], [0, 1, 127, 966, 1701, 1050, 266, 28, 1]]


In [8]:
t = StirlingSet.tab(9)
PrintTabl(t)

[1]
[0, 1]
[0, 1, 1]
[0, 1, 3, 1]
[0, 1, 7, 6, 1]
[0, 1, 15, 25, 10, 1]
[0, 1, 31, 90, 65, 15, 1]
[0, 1, 63, 301, 350, 140, 21, 1]
[0, 1, 127, 966, 1701, 1050, 266, 28, 1]


In [10]:
t = StirlingSet.tab(4)
PrintTerms(t)

0 [0, 0] 1
1 [1, 0] 0
2 [1, 1] 1
3 [2, 0] 0
4 [2, 1] 1
5 [2, 2] 1
6 [3, 0] 0
7 [3, 1] 1
8 [3, 2] 3
9 [3, 3] 1


In [28]:
t = StirlingSet.tab(9)
PrintRows(t)

|  Row   |  Seq   |
| :---   |  :---  |
| Row0 | [1] |
| Row1 | [0, 1] |
| Row2 | [0, 1, 1] |
| Row3 | [0, 1, 3, 1] |
| Row4 | [0, 1, 7, 6, 1] |
| Row5 | [0, 1, 15, 25, 10, 1] |
| Row6 | [0, 1, 31, 90, 65, 15, 1] |
| Row7 | [0, 1, 63, 301, 350, 140, 21, 1] |
| Row8 | [0, 1, 127, 966, 1701, 1050, 266, 28, 1] |


| Row   |  Seq   |
| :---  |  :---  |
| Row0 | [1] |
| Row1 | [0, 1] |
| Row2 | [0, 1, 1] |
| Row3 | [0, 1, 3, 1] |
| Row4 | [0, 1, 7, 6, 1] |
| Row5 | [0, 1, 15, 25, 10, 1] |
| Row6 | [0, 1, 31, 90, 65, 15, 1] |
| Row7 | [0, 1, 63, 301, 350, 140, 21, 1] |
| Row8 | [0, 1, 127, 966, 1701, 1050, 266, 28, 1] |

In [11]:
PrintRowArray(stirlingset, 6, 7)

| DiagRow  |   Seq  |
| :---     |  :---  |
| DiagRow0 | [1, 1, 1, 1, 1, 1, 1]|
| DiagRow1 | [0, 1, 3, 6, 10, 15, 21]|
| DiagRow2 | [0, 1, 7, 25, 65, 140, 266]|
| DiagRow3 | [0, 1, 15, 90, 350, 1050, 2646]|
| DiagRow4 | [0, 1, 31, 301, 1701, 6951, 22827]|
| DiagRow5 | [0, 1, 63, 966, 7770, 42525, 179487]|


| DiagRow  |   Seq  |
| :---     |  :---  |
| DiagRow0 | [1, 1, 1, 1, 1, 1, 1]|
| DiagRow1 | [0, 1, 3, 6, 10, 15, 21]|
| DiagRow2 | [0, 1, 7, 25, 65, 140, 266]|
| DiagRow3 | [0, 1, 15, 90, 350, 1050, 2646]|
| DiagRow4 | [0, 1, 31, 301, 1701, 6951, 22827]|
| DiagRow5 | [0, 1, 63, 966, 7770, 42525, 179487]|

In [20]:
PrintColArray(stirlingset, 6, 7) 

| DiagCol  |   Seq  |
| :---     |  :---  |
| DiagCol0 | [1, 0, 0, 0, 0, 0] |
| DiagCol1 | [1, 1, 1, 1, 1, 1] |
| DiagCol2 | [1, 3, 7, 15, 31, 63] |
| DiagCol3 | [1, 6, 25, 90, 301, 966] |
| DiagCol4 | [1, 10, 65, 350, 1701, 7770] |
| DiagCol5 | [1, 15, 140, 1050, 6951, 42525] |
| DiagCol6 | [1, 21, 266, 2646, 22827, 179487] |


| DiagCol  |   Seq  |
| :---     |  :---  |
| DiagCol0 | [1, 0, 0, 0, 0, 0] |
| DiagCol1 | [1, 1, 1, 1, 1, 1] |
| DiagCol2 | [1, 3, 7, 15, 31, 63] |
| DiagCol3 | [1, 6, 25, 90, 301, 966] |
| DiagCol4 | [1, 10, 65, 350, 1701, 7770] |
| DiagCol5 | [1, 15, 140, 1050, 6951, 42525] |
| DiagCol6 | [1, 21, 266, 2646, 22827, 179487] |

In [23]:
PrintPolyRowArray(stirlingset, 6, 7)

| PolyRow  |   Seq  |
| :---     |  :---  |
| PolyRow0 | [1, 1, 1, 1, 1, 1, 1] |
| PolyRow1 | [0, 1, 2, 3, 4, 5, 6] |
| PolyRow2 | [0, 2, 6, 12, 20, 30, 42] |
| PolyRow3 | [0, 5, 22, 57, 116, 205, 330] |
| PolyRow4 | [0, 15, 94, 309, 756, 1555, 2850] |
| PolyRow5 | [0, 52, 454, 1866, 5428, 12880, 26682] |


| PolyRow  |   Seq  |
| :---     |  :---  |
| PolyRow0 | [1, 1, 1, 1, 1, 1, 1] |
| PolyRow1 | [0, 1, 2, 3, 4, 5, 6] |
| PolyRow2 | [0, 2, 6, 12, 20, 30, 42] |
| PolyRow3 | [0, 5, 22, 57, 116, 205, 330] |
| PolyRow4 | [0, 15, 94, 309, 756, 1555, 2850] |
| PolyRow5 | [0, 52, 454, 1866, 5428, 12880, 26682] |

In [24]:
PrintPolyColArray(stirlingset, 6, 7)

| PolyCol  |   Seq  |
| :---     |  :---  |
| PolyCol0 | [1, 0, 0, 0, 0, 0, 0] |
| PolyCol1 | [1, 1, 2, 5, 15, 52, 203] |
| PolyCol2 | [1, 2, 6, 22, 94, 454, 2430] |
| PolyCol3 | [1, 3, 12, 57, 309, 1866, 12351] |
| PolyCol4 | [1, 4, 20, 116, 756, 5428, 42356] |
| PolyCol5 | [1, 5, 30, 205, 1555, 12880, 115155] |


| PolyCol  |   Seq  |
| :---     |  :---  |
| PolyCol0 | [1, 0, 0, 0, 0, 0, 0] |
| PolyCol1 | [1, 1, 2, 5, 15, 52, 203] |
| PolyCol2 | [1, 2, 6, 22, 94, 454, 2430] |
| PolyCol3 | [1, 3, 12, 57, 309, 1866, 12351] |
| PolyCol4 | [1, 4, 20, 116, 756, 5428, 42356] |
| PolyCol5 | [1, 5, 30, 205, 1555, 12880, 115155] |

In [13]:
PrintViews(Abel)

# Abel
['A061356', 'A137452', 'A139526']

Abel Triangle view
|  Row   |  Seq   |
| :---   |  :---  |
| Row0 | [1] |
| Row1 | [0, 1] |
| Row2 | [0, 2, 1] |
| Row3 | [0, 9, 6, 1] |
| Row4 | [0, 64, 48, 12, 1] |
| Row5 | [0, 625, 500, 150, 20, 1] |
| Row6 | [0, 7776, 6480, 2160, 360, 30, 1] |

Abel Triangles
| Flat       |  Seq  |
| :---       | :---  |
| Tabl       | [[1], [0, 1], [0, 2, 1], [0, 9, 6, 1], [0, 64, 48, 12, 1], [0, 625, 500, 150, 20, 1], [0, 7776, 6480, 2160, 360, 30, 1]] |
| RevTabl    | [[1], [1, 0], [1, 2, 0], [1, 6, 9, 0], [1, 12, 48, 64, 0], [1, 20, 150, 500, 625, 0], [1, 30, 360, 2160, 6480, 7776, 0]] |
| AntiDiag   | [[1], [0], [0, 1], [0, 2], [0, 9, 1], [0, 64, 6], [0, 625, 48, 1]] |
| AccTabl    | [[1], [0, 1], [0, 2, 3], [0, 9, 15, 16], [0, 64, 112, 124, 125], [0, 625, 1125, 1275, 1295, 1296], [0, 7776, 14256, 16416, 16776, 16806, 16807]] |
| RevAccTabl | [[1], [1, 0], [3, 2, 0], [16, 15, 9, 0], [125, 124, 112, 64, 0], [1296, 1295, 1275, 1125, 625, 0], [16807, 168


# Abel
['A061356', 'A137452', 'A139526']

Abel Triangle view

|  Row   |  Seq   |
| :---   |  :---  |
| Row0 | [1] |
| Row1 | [0, 1] |
| Row2 | [0, 2, 1] |
| Row3 | [0, 9, 6, 1] |
| Row4 | [0, 64, 48, 12, 1] |
| Row5 | [0, 625, 500, 150, 20, 1] |
| Row6 | [0, 7776, 6480, 2160, 360, 30, 1] |

Abel Triangles

| Flat       |  Seq  |
| :---       | :---  |
| Tabl       | [[1], [0, 1], [0, 2, 1], [0, 9, 6, 1], [0, 64, 48, 12, 1], [0, 625, 500, 150, 20, 1], [0, 7776, 6480, 2160, 360, 30, 1]] |
| RevTabl    | [[1], [1, 0], [1, 2, 0], [1, 6, 9, 0], [1, 12, 48, 64, 0], [1, 20, 150, 500, 625, 0], [1, 30, 360, 2160, 6480, 7776, 0]] |
| AntiDiag   | [[1], [0], [0, 1], [0, 2], [0, 9, 1], [0, 64, 6], [0, 625, 48, 1]] |
| AccTabl    | [[1], [0, 1], [0, 2, 3], [0, 9, 15, 16], [0, 64, 112, 124, 125], [0, 625, 1125, 1275, 1295, 1296], [0, 7776, 14256, 16416, 16776, 16806, 16807]] |
| RevAccTabl | [[1], [1, 0], [3, 2, 0], [16, 15, 9, 0], [125, 124, 112, 64, 0], [1296, 1295, 1275, 1125, 625, 0], [16807, 16806, 16776, 16416, 14256, 7776, 0]] |
| AccRevTabl | [[1], [1, 1], [1, 3, 3], [1, 7, 16, 16], [1, 13, 61, 125, 125], [1, 21, 171, 671, 1296, 1296], [1, 31, 391, 2551, 9031, 16807, 16807]] |

Abel Row sums

| Trait        |   Seq  |
| :---         |  :---  |
| RowSum       | [1, 1, 3, 16, 125, 1296, 16807] |
| EvenSum      | [1, 0, 1, 6, 49, 520, 6841] |
| OddSum       | [0, 1, 2, 10, 76, 776, 9966] |
| AltSum       | [1, -1, -1, -4, -27, -256, -3125] |
| AccSum       | [1, 1, 5, 40, 425, 5616, 88837] |
| AccRevSum    | [1, 2, 7, 40, 325, 3456, 45619] |
| AntiDiagSum  | [1, 0, 1, 2, 10, 70, 674] |

Abel Transforms

| Trans      |   Seq  |
| :---       |  :---  |
| RowLcm     | [1, 1, 2, 18, 192, 7500, 38880] |
| RowGcd     | [1, 1, 2, 3, 4, 5, 6] |
| RowMax     | [1, 1, 2, 9, 64, 625, 7776] |
| ColMiddle  | [1, 0, 2, 9, 48, 500, 2160] |
| ColECenter | [1, 2, 48, 2160] |
| ColOCenter | [0, 9, 500] |
| ColLeft    | [1, 0, 0, 0, 0, 0, 0] |
| ColRight   | [1, 1, 1, 1, 1, 1, 1] |
| TransSqrs  | [0, 1, 6, 42, 380, 4320, 59682] |
| TransNat0  | [0, 1, 4, 24, 200, 2160, 28812] |
| TransNat1  | [1, 2, 7, 40, 325, 3456, 45619] |

Abel Diagonals as rows

| DiagRow  |   Seq  |
| :---     |  :---  |
| DiagRow0 | [1, 1, 1, 1, 1, 1, 1]|
| DiagRow1 | [0, 2, 6, 12, 20, 30, 42]|
| DiagRow2 | [0, 9, 48, 150, 360, 735, 1344]|
| DiagRow3 | [0, 64, 500, 2160, 6860, 17920, 40824]|
| DiagRow4 | [0, 625, 6480, 36015, 143360, 459270, 1260000]|
| DiagRow5 | [0, 7776, 100842, 688128, 3306744, 12600000, 40584852]|
| DiagRow6 | [0, 117649, 1835008, 14880348, 84000000, 372027810, 1379524608]|

Abel Diagonals as columns

| DiagCol  |   Seq  |
| :---     |  :---  |
| DiagCol0 | [1, 0, 0, 0, 0, 0, 0] |
| DiagCol1 | [1, 2, 9, 64, 625, 7776, 117649] |
| DiagCol2 | [1, 6, 48, 500, 6480, 100842, 1835008] |
| DiagCol3 | [1, 12, 150, 2160, 36015, 688128, 14880348] |
| DiagCol4 | [1, 20, 360, 6860, 143360, 3306744, 84000000] |
| DiagCol5 | [1, 30, 735, 17920, 459270, 12600000, 372027810] |
| DiagCol6 | [1, 42, 1344, 40824, 1260000, 40584852, 1379524608] |

Abel Polynomial values as rows

| PolyRow  |   Seq  |
| :---     |  :---  |
| PolyRow0 | [1, 1, 1, 1, 1, 1, 1] |
| PolyRow1 | [0, 1, 2, 3, 4, 5, 6] |
| PolyRow2 | [0, 3, 8, 15, 24, 35, 48] |
| PolyRow3 | [0, 16, 50, 108, 196, 320, 486] |
| PolyRow4 | [0, 125, 432, 1029, 2048, 3645, 6000] |
| PolyRow5 | [0, 1296, 4802, 12288, 26244, 50000, 87846] |
| PolyRow6 | [0, 16807, 65536, 177147, 400000, 805255, 1492992] |

Abel Polynomial values as columns

| PolyCol  |   Seq  |
| :---     |  :---  |
| PolyCol0 | [1, 0, 0, 0, 0, 0, 0] |
| PolyCol1 | [1, 1, 3, 16, 125, 1296, 16807] |
| PolyCol2 | [1, 2, 8, 50, 432, 4802, 65536] |
| PolyCol3 | [1, 3, 15, 108, 1029, 12288, 177147] |
| PolyCol4 | [1, 4, 24, 196, 2048, 26244, 400000] |
| PolyCol5 | [1, 5, 35, 320, 3645, 50000, 805255] |
| PolyCol6 | [1, 6, 48, 486, 6000, 87846, 1492992] |

# TABLE TRANSFORMS

In [14]:
StirlingSet.revinv(9)

[[1],
 [1, 0],
 [1, -1, 0],
 [1, -3, 2, 0],
 [1, -6, 11, -6, 0],
 [1, -10, 35, -50, 24, 0],
 [1, -15, 85, -225, 274, -120, 0],
 [1, -21, 175, -735, 1624, -1764, 720, 0],
 [1, -28, 322, -1960, 6769, -13132, 13068, -5040, 0]]

'StirlingSet.invrev', i.e. the inverse of the reversed StirlingSet triangle does not exist because StirlingSet.rev is not invertible. In such cases simply the empty triangle [] is returned (and no error thrown).

In [15]:
StirlingSet.invrev(9)

[]

Conversely, a triangle which has not a 'revinv' but an 'invrev' form is the Delannoy triangle. Of course, all this depends on how the triangle was defined in the first place. And indeed some call the reversed Delannoy triangle the Delannoy triangle. 

In [16]:
Delannoy.invrev(9)

[[1],
 [-1, 1],
 [2, -3, 1],
 [-6, 10, -5, 1],
 [22, -38, 22, -7, 1],
 [-90, 158, -98, 38, -9, 1],
 [394, -698, 450, -194, 58, -11, 1],
 [-1806, 3218, -2126, 978, -334, 82, -13, 1],
 [8558, -15310, 10286, -4942, 1838, -526, 110, -15, 1]]

In [25]:
AntiDiagTabl(StirlingCycle.tab(10))

[[1],
 [0],
 [0, 1],
 [0, 1],
 [0, 2, 1],
 [0, 6, 3],
 [0, 24, 11, 1],
 [0, 120, 50, 6],
 [0, 720, 274, 35, 1],
 [0, 5040, 1764, 225, 10]]

In [26]:
AccTabl(Abel.tab(9))

[[1],
 [0, 1],
 [0, 2, 3],
 [0, 9, 15, 16],
 [0, 64, 112, 124, 125],
 [0, 625, 1125, 1275, 1295, 1296],
 [0, 7776, 14256, 16416, 16776, 16806, 16807],
 [0, 117649, 218491, 254506, 261366, 262101, 262143, 262144],
 [0, 2097152, 3932160, 4620288, 4763648, 4781568, 4782912, 4782968, 4782969]]

In [27]:
RevAccTabl(Abel.tab(9))

[[1],
 [1, 0],
 [3, 2, 0],
 [16, 15, 9, 0],
 [125, 124, 112, 64, 0],
 [1296, 1295, 1275, 1125, 625, 0],
 [16807, 16806, 16776, 16416, 14256, 7776, 0],
 [262144, 262143, 262101, 261366, 254506, 218491, 117649, 0],
 [4782969, 4782968, 4782912, 4781568, 4763648, 4620288, 3932160, 2097152, 0]]

In [28]:
AccRevTabl(Abel.tab(9))

[[1],
 [1, 1],
 [1, 3, 3],
 [1, 7, 16, 16],
 [1, 13, 61, 125, 125],
 [1, 21, 171, 671, 1296, 1296],
 [1, 31, 391, 2551, 9031, 16807, 16807],
 [1, 43, 778, 7638, 43653, 144495, 262144, 262144],
 [1, 57, 1401, 19321, 162681, 850809, 2685817, 4782969, 4782969]]

In [29]:
PrintTabls(Abel)

# Abel : Tables
| Trait    |   Seq  |
| :---     |  :---  |
| FlatTabl        | [1, 0, 1, 0, 2, 1, 0, 9, 6, 1, 0, 64, 48, 12, 1, 0, 625, 500, 150, 20, 1, 0, 7776, 6480, 2160, 360, 30, 1, 0, 117649, 100842, 36015, 6860, 735, 42, 1] |
| FlatRevTabl     | [1, 1, 0, 1, 2, 0, 1, 6, 9, 0, 1, 12, 48, 64, 0, 1, 20, 150, 500, 625, 0, 1, 30, 360, 2160, 6480, 7776, 0, 1, 42, 735, 6860, 36015, 100842, 117649, 0] |
| FlatInvTabl     | [1, 0, 1, 0, -2, 1, 0, 3, -6, 1, 0, -4, 24, -12, 1, 0, 5, -80, 90, -20, 1, 0, -6, 240, -540, 240, -30, 1, 0, 7, -672, 2835, -2240, 525, -42, 1] |
| FlatInvRevTabl  | [] |
| FlatRevInvTabl  | [1, 1, 0, 1, -2, 0, 1, -6, 3, 0, 1, -12, 24, -4, 0, 1, -20, 90, -80, 5, 0, 1, -30, 240, -540, 240, -6, 0, 1, -42, 525, -2240, 2835, -672, 7, 0] |
| FlatAccTabl     | [1, 0, 1, 0, 2, 3, 0, 9, 15, 16, 0, 64, 112, 124, 125, 0, 625, 1125, 1275, 1295, 1296, 0, 7776, 14256, 16416, 16776, 16806, 16807, 0, 117649, 218491, 254506, 261366, 262101, 262143, 262144] |
| FlatRevAccTabl  | [1, 1

# Abel : Tables
| Trait    |   Seq  |
| :---     |  :---  |
| FlatTabl        | [1, 0, 1, 0, 2, 1, 0, 9, 6, 1, 0, 64, 48, 12, 1, 0, 625, 500, 150, 20, 1, 0, 7776, 6480, 2160, 360, 30, 1, 0, 117649, 100842, 36015, 6860, 735, 42, 1] |
| FlatRevTabl     | [1, 1, 0, 1, 2, 0, 1, 6, 9, 0, 1, 12, 48, 64, 0, 1, 20, 150, 500, 625, 0, 1, 30, 360, 2160, 6480, 7776, 0, 1, 42, 735, 6860, 36015, 100842, 117649, 0] |
| FlatInvTabl     | [1, 0, 1, 0, -2, 1, 0, 3, -6, 1, 0, -4, 24, -12, 1, 0, 5, -80, 90, -20, 1, 0, -6, 240, -540, 240, -30, 1, 0, 7, -672, 2835, -2240, 525, -42, 1] |
| FlatInvRevTabl  | [] |
| FlatRevInvTabl  | [1, 1, 0, 1, -2, 0, 1, -6, 3, 0, 1, -12, 24, -4, 0, 1, -20, 90, -80, 5, 0, 1, -30, 240, -540, 240, -6, 0, 1, -42, 525, -2240, 2835, -672, 7, 0] |
| FlatAccTabl     | [1, 0, 1, 0, 2, 3, 0, 9, 15, 16, 0, 64, 112, 124, 125, 0, 625, 1125, 1275, 1295, 1296, 0, 7776, 14256, 16416, 16776, 16806, 16807, 0, 117649, 218491, 254506, 261366, 262101, 262143, 262144] |
| FlatRevAccTabl  | [1, 1, 0, 3, 2, 0, 16, 15, 9, 0, 125, 124, 112, 64, 0, 1296, 1295, 1275, 1125, 625, 0, 16807, 16806, 16776, 16416, 14256, 7776, 0, 262144, 262143, 262101, 261366, 254506, 218491, 117649, 0] |
| FlatAccRevTabl  | [1, 1, 1, 1, 3, 3, 1, 7, 16, 16, 1, 13, 61, 125, 125, 1, 21, 171, 671, 1296, 1296, 1, 31, 391, 2551, 9031, 16807, 16807, 1, 43, 778, 7638, 43653, 144495, 262144, 262144] |
| FlatAntiDiagTabl | [1, 0, 0, 1, 0, 2, 0, 9, 1, 0, 64, 6, 0, 625, 48, 1, 0, 7776, 500, 12] |